Build failure: python3.12-certbot-dns-cloudflare-3.1.0

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

Update to the latest version and see the error.

### Can Hydra reproduce this build failure?

Please select the Hydra Status.

### Link to Hydra build job

_No response_

### Relevant log output

```console
Sourcing python-remove-tests-dir-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing python-remove-bin-bytecode-hook.sh
Sourcing setuptools-build-hook
Using setuptoolsBuildPhase
Sourcing pypa-install-hook
Using pypaInstallPhase
Sourcing python-imports-check-hook.sh
Using pythonImportsCheckPhase
Sourcing python-namespaces-hook
Sourcing python-catch-conflicts-hook.sh
Sourcing pytest-check-hook
Using pytestCheckPhase
Running phase: unpackPhase
unpacking source archive /nix/store/m8lwh73rfapppr6sv33mh20b3gmh68di-source
source root is source/certbot-dns-cloudflare
setting SOURCE_DATE_EPOCH to timestamp 315619200 of file "source/certbot-dns-cloudflare/setup.py"
Running phase: patchPhase
Running phase: updateAutotoolsGnuConfigScriptsPhase
Running phase: configurePhase
no configure script, doing nothing
Running phase: buildPhase
Executing setuptoolsBuildPhase
setup.py build flags: ''
running bdist_wheel
running build
running build_py
creating build/lib/certbot_dns_cloudflare
copying certbot_dns_cloudflare/__init__.py -> build/lib/certbot_dns_cloudflare
creating build/lib/certbot_dns_cloudflare/_internal
copying certbot_dns_cloudflare/_internal/__init__.py -> build/lib/certbot_dns_cloudflare/_internal
copying certbot_dns_cloudflare/_internal/dns_cloudflare.py -> build/lib/certbot_dns_cloudflare/_internal
creating build/lib/certbot_dns_cloudflare/_internal/tests
copying certbot_dns_cloudflare/_internal/tests/__init__.py -> build/lib/certbot_dns_cloudflare/_internal/tests
copying certbot_dns_cloudflare/_internal/tests/dns_cloudflare_test.py -> build/lib/certbot_dns_cloudflare/_internal/tests
running egg_info
creating certbot_dns_cloudflare.egg-info
writing certbot_dns_cloudflare.egg-info/PKG-INFO
writing dependency_links to certbot_dns_cloudflare.egg-info/dependency_links.txt
writing entry points to certbot_dns_cloudflare.egg-info/entry_points.txt
writing requirements to certbot_dns_cloudflare.egg-info/requires.txt
writing top-level names to certbot_dns_cloudflare.egg-info/top_level.txt
writing manifest file 'certbot_dns_cloudflare.egg-info/SOURCES.txt'
reading manifest file 'certbot_dns_cloudflare.egg-info/SOURCES.txt'
reading manifest template 'MANIFEST.in'
warning: no previously-included files matching '__pycache__' found anywhere in distribution
warning: no previously-included files matching '*.py[cod]' found anywhere in distribution
adding license file 'LICENSE.txt'
writing manifest file 'certbot_dns_cloudflare.egg-info/SOURCES.txt'
copying certbot_dns_cloudflare/py.typed -> build/lib/certbot_dns_cloudflare
/nix/store/awqs4f1vqa4z83rvngh58hncqhvafymm-python3.12-setuptools-75.8.0/lib/python3.12/site-packages/setuptools/_distutils/cmd.py:79: SetuptoolsDeprecationWarning: setup.py install is deprecated.
!!

        ********************************************************************************
        Please avoid running ``setup.py`` directly.
        Instead, use pypa/build, pypa/installer or other
        standards-based tools.

        See https://blog.ganssle.io/articles/2021/10/setup-py-deprecated.html for details.
        ********************************************************************************

!!
  self.initialize_options()
installing to build/bdist.linux-x86_64/wheel
running install
running install_lib
creating build/bdist.linux-x86_64/wheel
creating build/bdist.linux-x86_64/wheel/certbot_dns_cloudflare
copying build/lib/certbot_dns_cloudflare/__init__.py -> build/bdist.linux-x86_64/wheel/./certbot_dns_cloudflare
creating build/bdist.linux-x86_64/wheel/certbot_dns_cloudflare/_internal
copying build/lib/certbot_dns_cloudflare/_internal/__init__.py -> build/bdist.linux-x86_64/wheel/./certbot_dns_cloudflare/_internal
copying build/lib/certbot_dns_cloudflare/_internal/dns_cloudflare.py -> build/bdist.linux-x86_64/wheel/./certbot_dns_cloudflare/_internal
creating build/bdist.linux-x86_64/wheel/certbot_dns_cloudflare/_internal/tests
copying build/lib/certbot_dns_cloudflare/_internal/tests/__init__.py -> build/bdist.linux-x86_64/wheel/./certbot_dns_cloudflare/_internal/tests
copying build/lib/certbot_dns_cloudflare/_internal/tests/dns_cloudflare_test.py -> build/bdist.linux-x86_64/wheel/./certbot_dns_cloudflare/_internal/tests
copying build/lib/certbot_dns_cloudflare/py.typed -> build/bdist.linux-x86_64/wheel/./certbot_dns_cloudflare
running install_egg_info
Copying certbot_dns_cloudflare.egg-info to build/bdist.linux-x86_64/wheel/./certbot_dns_cloudflare-3.1.0-py3.12.egg-info
running install_scripts
creating build/bdist.linux-x86_64/wheel/certbot_dns_cloudflare-3.1.0.dist-info/WHEEL
creating 'dist/certbot_dns_cloudflare-3.1.0-py3-none-any.whl' and adding 'build/bdist.linux-x86_64/wheel' to it
adding 'certbot_dns_cloudflare/__init__.py'
adding 'certbot_dns_cloudflare/py.typed'
adding 'certbot_dns_cloudflare/_internal/__init__.py'
adding 'certbot_dns_cloudflare/_internal/dns_cloudflare.py'
adding 'certbot_dns_cloudflare/_internal/tests/__init__.py'
adding 'certbot_dns_cloudflare/_internal/tests/dns_cloudflare_test.py'
adding 'certbot_dns_cloudflare-3.1.0.dist-info/LICENSE.txt'
adding 'certbot_dns_cloudflare-3.1.0.dist-info/METADATA'
adding 'certbot_dns_cloudflare-3.1.0.dist-info/WHEEL'
adding 'certbot_dns_cloudflare-3.1.0.dist-info/entry_points.txt'
adding 'certbot_dns_cloudflare-3.1.0.dist-info/top_level.txt'
adding 'certbot_dns_cloudflare-3.1.0.dist-info/RECORD'
removing build/bdist.linux-x86_64/wheel
Finished executing setuptoolsBuildPhase
Running phase: installPhase
Executing pypaInstallPhase
Successfully installed certbot_dns_cloudflare-3.1.0-py3-none-any.whl
Finished executing pypaInstallPhase
Running phase: pythonOutputDistPhase
Executing pythonOutputDistPhase
Finished executing pythonOutputDistPhase
Running phase: fixupPhase
shrinking RPATHs of ELF executables and libraries in /nix/store/slr527lmhqzrynk0zb566aib52ag1s5h-python3.12-certbot-dns-cloudflare-3.1.0
checking for references to /build/ in /nix/store/slr527lmhqzrynk0zb566aib52ag1s5h-python3.12-certbot-dns-cloudflare-3.1.0...
patching script interpreter paths in /nix/store/slr527lmhqzrynk0zb566aib52ag1s5h-python3.12-certbot-dns-cloudflare-3.1.0
stripping (with command strip and flags -S -p) in  /nix/store/slr527lmhqzrynk0zb566aib52ag1s5h-python3.12-certbot-dns-cloudflare-3.1.0/lib
shrinking RPATHs of ELF executables and libraries in /nix/store/cdfk1inm8alh20g7f85cxa6xlnah523y-python3.12-certbot-dns-cloudflare-3.1.0-dist
checking for references to /build/ in /nix/store/cdfk1inm8alh20g7f85cxa6xlnah523y-python3.12-certbot-dns-cloudflare-3.1.0-dist...
patching script interpreter paths in /nix/store/cdfk1inm8alh20g7f85cxa6xlnah523y-python3.12-certbot-dns-cloudflare-3.1.0-dist
Executing pythonRemoveTestsDir
Finished executing pythonRemoveTestsDir
Running phase: installCheckPhase
no Makefile or custom installCheckPhase, doing nothing
Running phase: pythonCatchConflictsPhase
Running phase: pythonRemoveBinBytecodePhase
Running phase: pythonImportsCheckPhase
Executing pythonImportsCheckPhase
Running phase: pytestCheckPhase
Executing pytestCheckPhase
pytest flags: -m pytest -p no:cacheprovider -W ignore::DeprecationWarning
============================= test session starts ==============================
platform linux -- Python 3.12.8, pytest-8.3.4, pluggy-1.5.0
rootdir: /build/source
configfile: pytest.ini
plugins: anyio-4.8.0
collected 0 items / 1 error                                                    

==================================== ERRORS ====================================
_ ERROR collecting certbot-dns-cloudflare/certbot_dns_cloudflare/_internal/tests/dns_cloudflare_test.py _
ImportError while importing test module '/build/source/certbot-dns-cloudflare/certbot_dns_cloudflare/_internal/tests/dns_cloudflare_test.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/nix/store/0l539chjmcq5kdd43j6dgdjky4sjl7hl-python3-3.12.8/lib/python3.12/importlib/__init__.py:90: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
certbot_dns_cloudflare/_internal/tests/dns_cloudflare_test.py:7: in <module>
    import CloudFlare
E   ModuleNotFoundError: No module named 'CloudFlare'
=========================== short test summary info ============================
ERROR certbot_dns_cloudflare/_internal/tests/dns_cloudflare_test.py
!!!!!!!!!!!!!!!!!!!! Interrupted: 1 error during collection !!!!!!!!!!!!!!!!!!!!
=============================== 1 error in 0.10s ===============================
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.13.1, NixOS, 25.05 (Warbler), 25.05.20250204.799ba5b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.12`
 - nixpkgs: `/home/jflabonte/.nix-defexpr/channels/nixpkgs`



### Notify maintainers


---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/development/python-modules/certbot-dns-cloudflare/default.nix
+++ b/pkgs/development/python-modules/certbot-dns-cloudflare/default.nix
@@ -37,5 +37,7 @@ buildPythonPackage rec {
 
   meta = certbot.meta // {
     description = "Cloudflare DNS Authenticator plugin for Certbot";
+    # https://github.com/certbot/certbot/pull/10182
+    broken = true;
   };
 }

--- a/pkgs/development/python-modules/certbot-dns-cloudflare/default.nix
+++ b/pkgs/development/python-modules/certbot-dns-cloudflare/default.nix
@@ -5,18 +5,21 @@
   cloudflare,
   pytestCheckHook,
   pythonOlder,
+  setuptools,
 }:
 
 buildPythonPackage rec {
   pname = "certbot-dns-cloudflare";
-  format = "setuptools";
+  pyproject = true;
 
   inherit (certbot) src version;
-  disabled = pythonOlder "3.6";
+  disabled = pythonOlder "3.9";
 
   sourceRoot = "${src.name}/certbot-dns-cloudflare";
 
-  propagatedBuildInputs = [
+  build-system = [ setuptools ];
+
+  dependencies = [
     acme
     certbot
     cloudflare

